### PR TITLE
Release 3.0.2 take2

### PR DESCRIPTION
**Pull request by GodloveD**
It can (and has) been argued that every PR since 3.0.1 has been a bug fix PR.  So I've just merged it all and called it 3.0.2.  Would like another set of eyes just to make sure that this is sane.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+# Changes Since v3.0.2
+
+# v3.0.2 - []
+
+  - Added build requirements to RPM spec file for easier RPM building
+  - Replaced the `$SINGULARITY_NOHTTPS` environment variable to allow pulls from http instead of https
+  - Replaced the `$SINGULARITY_TMPDIR` environment variable for builds performed by the build, actions and pull commands
+  - Added the ability to pull from a private shub registry
+  - Replaced the `$SINGULARITY_DOCKER_USERNAME` and `$SINGULARITY_DOCKER_PASSWORD` environment variables for authentication and introduced the `--docker-login` flag for interactive authentication
+  - Made the following packages open:
```
Comment (bauerm97) [inline CHANGELOG.md]: This should not be documented in the changelog. This is already documented in the commit log, and is too granular for a changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+    - `pkg/client/library`
+    - `pkg/client/shub`
+    - `pkg/signing`
+    - `pkg/sypgp`
+    - `pkg/util/loop`
+    - `pkg/util/user-agent`
+  - Added support for pulling directly from http and https
+  - Add recipe file for packaging Singularity on alpine
+  - Added option for a user to replicate the 2.x behavior of `--net` with the `--network=none` option/argument pair
+  - Change error code to 1 on invalid/incomplete commands
```
Comment (bauerm97) [inline CHANGELOG.md]: Better line:

`Commands now correctly return 1 when incorrectly ran without arguments`

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Made the following packages open:
+    - `pkg/client/library`
+    - `pkg/client/shub`
+    - `pkg/signing`
+    - `pkg/sypgp`
+    - `pkg/util/loop`
+    - `pkg/util/user-agent`
+  - Added support for pulling directly from http and https
+  - Add recipe file for packaging Singularity on alpine
+  - Added option for a user to replicate the 2.x behavior of `--net` with the `--network=none` option/argument pair
```
Comment (bauerm97) [inline CHANGELOG.md]: Better line:

```Running a container with `--network="none"` no longer incorrectly fails with an error message when```

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Change error code to 1 on invalid/incomplete commands 
+  - Silence progress bars when pushing or pulling with the `--quiet` or `--silent` flags
```
Comment (bauerm97) [inline CHANGELOG.md]: I don't think we need this line, but I'm not opposed if you really want it. I would change it to:

```Progress bars no longer incorrectly display when running with `--quiet` or `--silent` ```

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Display contents of `91-environment.sh` file if it exists when `inspect --environment` is invoked
```
Comment (bauerm97) [inline CHANGELOG.md]: I don't think we need this line here

File: CHANGELOG.md [hunk 6]
```diff
@@ -86,4 +107,4 @@ _The old changelog can be found in the `release-2.6` branch_
     - `oci-archive`
   - The `from` keyword in the definition file now correctly parses a `docker` URI which includes the `registry` and/or `namespace` components
   - The `registry` and `namespace` keywords in the definition file are no longer supported. Instead, those values may all go into the `from` keyword
-  - Building from a tar archive of a `sandbox` no longer works
\ No newline at end of file
+  - Building from a tar archive of a `sandbox` no longer works
```
Comment (bauerm97) [inline CHANGELOG.md:101]: Is something actually different here, or is git just acting strange?

File: CHANGELOG.md [hunk 7]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Added the ability to pull from a private shub registry
+  - Replaced the `$SINGULARITY_DOCKER_USERNAME` and `$SINGULARITY_DOCKER_PASSWORD` environment variables for authentication and introduced the `--docker-login` flag for interactive authentication
+  - Made the following packages open:
+    - `pkg/client/library`
+    - `pkg/client/shub`
+    - `pkg/signing`
+    - `pkg/sypgp`
+    - `pkg/util/loop`
+    - `pkg/util/user-agent`
+  - Added support for pulling directly from http and https
```
Comment (bauerm97) [inline CHANGELOG.md]: HTTP and HTTPS

File: CHANGELOG.md [hunk 8]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
 
 _The old changelog can be found in the `release-2.6` branch_
 
-# Changes Since v3.0.1
+# Changes Since v3.0.2
+
+# v3.0.2 - []
+
+  - Added build requirements to RPM spec file for easier RPM building
+  - Replaced the `$SINGULARITY_NOHTTPS` environment variable to allow pulls from http instead of https
```
Comment (bauerm97) [inline CHANGELOG.md]: ```The `$SINGULARITY_NOHTTPS`,`$SINGULARITY_TMPDIR`, and `$SINGULARITY_DOCKER_USERNAME`/`$SINGULARITY_DOCKER_PASSWORD` environment variables are now correctly respected```

File: CHANGELOG.md [hunk 9]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Replaced the `$SINGULARITY_TMPDIR` environment variable for builds performed by the build, actions and pull commands
+  - Added the ability to pull from a private shub registry
+  - Replaced the `$SINGULARITY_DOCKER_USERNAME` and `$SINGULARITY_DOCKER_PASSWORD` environment variables for authentication and introduced the `--docker-login` flag for interactive authentication
```
Comment (bauerm97) [inline CHANGELOG.md]: ```Added the `--docker-login` flag to enable interactive authentication with docker registries```

File: CHANGELOG.md [hunk 10]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
 
-# Changes Since v3.0.1
+# Changes Since v3.0.2
+
+# v3.0.2 - []
+
+  - Added build requirements to RPM spec file for easier RPM building
+  - Replaced the `$SINGULARITY_NOHTTPS` environment variable to allow pulls from http instead of https
+  - Replaced the `$SINGULARITY_TMPDIR` environment variable for builds performed by the build, actions and pull commands
+  - Added the ability to pull from a private shub registry
```
Comment (bauerm97) [inline CHANGELOG.md]: `Pulling from a private shub registry now works as expected`

File: CHANGELOG.md [hunk 11]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Replaced the `$SINGULARITY_DOCKER_USERNAME` and `$SINGULARITY_DOCKER_PASSWORD` environment variables for authentication and introduced the `--docker-login` flag for interactive authentication
+  - Made the following packages open:
+    - `pkg/client/library`
+    - `pkg/client/shub`
+    - `pkg/signing`
+    - `pkg/sypgp`
+    - `pkg/util/loop`
+    - `pkg/util/user-agent`
+  - Added support for pulling directly from http and https
+  - Add recipe file for packaging Singularity on alpine
```
Comment (bauerm97) [inline CHANGELOG.md]: Remove this line, covered by first comment.

File: CHANGELOG.md [hunk 12]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
 
 _The old changelog can be found in the `release-2.6` branch_
 
-# Changes Since v3.0.1
+# Changes Since v3.0.2
+
+# v3.0.2 - []
+
+  - Added build requirements to RPM spec file for easier RPM building
```
Comment (bauerm97) [inline CHANGELOG.md]: `Made minor improvements to RPM packaging and added basic support for alpine packaging`

File: CHANGELOG.md [hunk 13]
```diff
@@ -9,7 +9,28 @@ _With the release of `v3.0.0`, we're introducing a new changelog format in an at
...
+  - Replaced the `$SINGULARITY_NOHTTPS` environment variable to allow pulls from http instead of https
+  - Replaced the `$SINGULARITY_TMPDIR` environment variable for builds performed by the build, actions and pull commands
```
Comment (bauerm97) [inline CHANGELOG.md]: Remove this line

**Review by bauerm97 — CHANGES_REQUESTED** [review]
Just a few tweaks to make on the `CHANGELOG.md` before it's ready

File: CHANGELOG.md (hunk 4 above)
Comment (GodloveD) [reply CHANGELOG.md]: I'm happy with the change.  It's a very minor behavior change, but it is a behavior change so I think we need to document it.

File: CHANGELOG.md (hunk 5 above)
Comment (GodloveD) [reply CHANGELOG.md]: It's a very minor behavior change on a feature that is probably not used by many people.  But it's a behavior change so I believe it needs to be documented.

File: CHANGELOG.md (hunk 6 above)
Comment (GodloveD) [reply CHANGELOG.md:101]: I don't see a difference.  🤷‍♂️

File: CHANGELOG.md (hunk 11 above)
Comment (GodloveD) [reply CHANGELOG.md]: They are actually 2 seperate things afaik.  alpine != rpm

File: CHANGELOG.md (hunk 6 above)
Comment (bauerm97) [reply CHANGELOG.md:101]: Same, was wondering if you saw one ¯\\\_(ツ)\_/¯

File: CHANGELOG.md (hunk 5 above)
Comment (bauerm97) [reply CHANGELOG.md]: OK, if we want to leave this in then:

```Contents of `91-environment.sh` file are now displayed if appropriate when running `inspect --environment` ```

**Review by bauerm97 — APPROVED** [review]
Looks ready to tag an `rc`. We need to make sure to remember to add a date in the changelog before we tag the actual release.

**Pull request merged**